drf_extra_fields/fields.py

```diff
@@ -59,32 +53,63 @@ def to_internal_value(self, base64_data):
             try:
                 decoded_file = base64.b64decode(base64_data)
             except (TypeError, binascii.Error):
-                raise ValidationError(_("Please upload a valid image."))
+                raise ValidationError(self.INVALID_FILE_MESSAGE)
             # Generate file name:
             file_name = str(uuid.uuid4())[:12]  # 12 characters are more than enough.
             # Get the file name extension:
             file_extension = self.get_file_extension(file_name, decoded_file)
-            if file_extension not in ALLOWED_IMAGE_TYPES:
-                raise ValidationError(_("The type of the image couldn't be determined."))
+            if file_extension not in self.ALLOWED_TYPES:
+                raise ValidationError(self.INVALID_TYPE_MESSAGE)
             complete_file_name = file_name + "." + file_extension
             data = ContentFile(decoded_file, name=complete_file_name)
-            return super(Base64ImageField, self).to_internal_value(data)
+            return super(Base64FieldMixin, self).to_internal_value(data)
         raise ValidationError(_('This is not an base64 string'))
 
     def get_file_extension(self, filename, decoded_file):
-        extension = imghdr.what(filename, decoded_file)
-        extension = "jpg" if extension == "jpeg" else extension
-        return extension
+        raise NotImplemented
 
     def to_representation(self, image):
         if self.represent_in_base64:
             try:
                 with open(image.path, 'rb') as f:
                     return base64.b64encode(f.read()).decode()
             except Exception:
-                raise IOError("Error encoding image file")
+                raise IOError("Error encoding file")
         else:
-            return super(ImageField, self).to_representation(image)
+            return super(Base64FieldMixin, self).to_representation(image)
+
+
+class Base64ImageField(Base64FieldMixin, ImageField):
+    """
+    A django-rest-framework field for handling image-uploads through raw post data.
+    It uses base64 for en-/decoding the contents of the file.
+    """
+    ALLOWED_TYPES = (
+        "jpeg",
+        "jpg",
+        "png",
+        "gif"
+    )
+    INVALID_FILE_MESSAGE = _("Please upload a valid image.")
+    INVALID_TYPE_MESSAGE = _("The type of the image couldn't be determined.")
+
+    def get_file_extension(self, filename, decoded_file):
+        extension = imghdr.what(filename, decoded_file)
+        extension = "jpg" if extension == "jpeg" else extension
+        return extension
+
+
+class Base64FileField(Base64FieldMixin, FileField):
+    """
+    A django-rest-framework field for handling file-uploads through raw post data.
+    It uses base64 for en-/decoding the contents of the file.
+    """
+    ALLOWED_TYPES = NotImplemented
```

Comment:
We can add a helper message to the Not implemented error in order to provide better information to the programmer.